include/rtc/pacinghandler.hpp

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2020 Staz Modrzynski
+ * Copyright (c) 2020 Paul-Louis Ageneau
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef RTC_PACING_HANDLER_H
+#define RTC_PACING_HANDLER_H
+
+#if RTC_ENABLE_MEDIA
+
+#include "mediahandler.hpp"
+#include "utils.hpp"
+
+#include <atomic>
+#include <deque>
+
+namespace rtc {
+
+// Paced sending of RTP packets. Takes a stream of RTP packets that can an
+// uneven bitrate. It then delivers these packets in a smoother manner by
+// sending a fixed size of them on an interval
+class RTC_CPP_EXPORT PacingHandler : public MediaHandler {
+public:
+	static constexpr float DefaultOverageFactor = 2.0;
+	PacingHandler(float bytesPerMillisecond, std::chrono::milliseconds sendInterval,
+	              float overageFactor = DefaultOverageFactor);
+
+	void outgoing(message_vector &messages, const message_callback &send) override;
+
+private:
+	float mBytesPerMillisecond;
+	std::chrono::milliseconds mSendInterval;
+
+	std::atomic<bool> mHaveScheduled = false;
+	std::chrono::time_point<std::chrono::high_resolution_clock> mLastRun;
+
+	size_t mOverage;
+	float mOverageFactor;
+
+	class RTC_CPP_EXPORT Storage {
+	public:
+		std::mutex mMutex;
+		std::deque<message_ptr> mRtpBuffer = {};
```

Comment:
Nitpick, but you could use `queue` instead of `dequeue` as you don't need sequential access, only FIFO.

Reply:
Done!